lib/connections/limiter.go

```diff
@@ -155,34 +123,35 @@ func (lim *limiter) processDevicesConfiguration(from, to *config.Configuration)
 	}
 
 	// Delete remote devices which were removed in new configuration
-	for deviceID, shouldBeRemoved := range devicesToRemove {
-		l.Debugf("deviceID: %s, shouldBeRemoved: %t", deviceID, shouldBeRemoved)
-		if shouldBeRemoved {
-			lim.deleteDeviceLimiters(deviceID)
+	for _, v := range from.Devices {
+		if _, ok := seen[v.DeviceID]; !ok {
+			l.Debugf("deviceID: %s should be removed", v.DeviceID)
+			lim.deleteDeviceLimiters(v.DeviceID)
 		}
 	}
 
 	l.Debugln("Rebuild of device limiters map finished")
 }
 
 // Compare read/write limits in configurations
-func (lim *limiter) checkDeviceLimits(from, to config.Configuration) bool {
+func deviceLimitsChanged(from, to config.Configuration) bool {
 	if len(from.Devices) != len(to.Devices) {
-		return false
+		return true
 	}
+	fromMap := config.MapDeviceConfigs(from.Devices)
+	toMap := config.MapDeviceConfigs(to.Devices)
 	// len(from.Devices) == len(to.Devices) so we can do range from.Devices
-	for i := range from.Devices {
-		if from.Devices[i].DeviceID != to.Devices[i].DeviceID {
-			// Something has changed in device configuration
-			return false
+	for k := range fromMap {
+		if _, ok := toMap[k]; !ok {
+			return true
 		}
 		// Read/write limits were changed for this device
-		if from.Devices[i].MaxSendKbps != to.Devices[i].MaxSendKbps ||
-			from.Devices[i].MaxRecvKbps != to.Devices[i].MaxRecvKbps {
-			return false
+		if fromMap[k].MaxSendKbps != toMap[k].MaxSendKbps ||
+			fromMap[k].MaxRecvKbps != toMap[k].MaxRecvKbps {
```

Comment:
No manual line breaks (I was/am a fan of them too, but it's not the Syncthing/Go style ;) ).